public/static/js/spawned_hosts.js

```diff
@@ -92,6 +70,68 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope','$window', '$timeout', 'mciSp
     }
 
 
+
+    $scope.updateSpawnedHosts = function() {
+        mciSpawnRestService.getSpawnedHosts(
+            'hosts', {}, {
+                success: function(resp) {
+                    var hosts = resp.data;
+                    _.each(hosts, function(host) {
+                        for(var i = 0; i < $scope.hosts.length; i++) {
+                            if ($scope.hosts[i].id !== host.id && $scope.hosts[i].id !== host.tag) {
+                                continue;
+                            }
+                            $scope.computeUptime(host);
+                            $scope.computeExpirationTimes(host);
+                            $scope.hosts[i].uptime = host.uptime;
+                            $scope.hosts[i].expires_in = host.expires_in;
+                            $scope.hosts[i].status = host.status;
+                            $scope.hosts[i].id = host.id;
+                            if ($scope.hosts[i].instance_type === undefined || $scope.hosts[i].instance_type === "") {
+                                $scope.hosts[i].instance_type = host.instance_type;
+                            }
+                            if ($scope.hosts[i].selectedInstanceType === undefined || $scope.hosts[i].selectedInstanceType === "") {
+                                $scope.hosts[i].selectedInstanceType = host.instance_type;
+                            }
+                        }
+                    });
+                }
+            }
+        )
+    }
+
+    $scope.computeExpirationTimes = function(host) {
+            if (!host.isTerminated && +new Date(host.expiration_time) > +new Date("0001-01-01T00:00:00Z")) {
```

Comment:
You don't need the + there - javascript should handle the date comparisons correctly

Reply:
I think you need to take both of them out for this to work